### ISSUE TITLE
feat(gateway): expose opt-in guarded native inference receipts

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#9070]
Parent: #8972

## Current state

The production fak serve path already times native prefill and decode and holds the full logits used for each token choice, but those facts stop at an operator log. ChatResponse and agent.Completion expose neither generated token ids nor per-step chosen-token scores. A docs-only #8972 runner therefore cannot prove 64 selected-token normalized logits or true phase timing through the guarded HTTP path; direct GGUF loading would bypass serve and is invalid.

## Value

- For: native-model benchmark and regression harnesses that must prove which fak execution produced a response.
- Problem: the guarded endpoint returns text and aggregate usage but not the token-level and phase evidence needed for a quality-complete receipt.
- Today: scrape logs, fabricate phase splits, or call model.Session directly; none binds the evidence to the adjudicated production request.
- Better because: an explicit request opt-in returns a typed fak extension derived inside the same InKernelPlanner decode that served the response.
- Witness: a deterministic in-kernel test proves emitted token ids and chosen-token log_softmax values come from the actual prefill/step logits, phase durations are real nonnegative measurements, and backend/forward identity is present.

## Problem frame

- Centrality: Enabling; this supplies the missing native evidence seam for #8972.
- P1: advanced; token and timing facts are captured at their source.
- P2: advanced; prefill and decode remain separately measured rather than split after the fact.
- P3: preserved; only the fak-native planner can populate the receipt and no external-engine fallback is introduced.
- P4: advanced; the response becomes machine-readable and reusable by oracle bundles.

## In scope

Add an opt-in request carrier, thread it through SampleOpt, capture generated token ids and the finite selected-token log probability from the real logits at each decode choice, attach exact prefill and decode seconds plus native execution identity to agent.Completion, and project it under ChatResponse.fak. Normal responses remain byte-compatible when the opt-in is absent.

For this first spine, define selected-token score as log_softmax of the unmodified model logits for the chosen token. The #8972 campaign uses temperature zero with no bias or repetition penalties. If nonzero sampling modifiers make the score ambiguous, fail closed or clearly mark the unsupported envelope rather than guess.

## Out of scope

Full-vocabulary logit tensors, streaming receipts, external provider logprobs, RSS sampling, benchmark artifact capture, or performance optimization.

## Done condition

A buffered POST /v1/chat/completions request can explicitly ask for a native receipt and receive exactly one token id and one finite selected-token log probability per generated token, true prefill/decode durations, model/backend/forward-path/q4k identity, and no receipt on the default path. The values are produced by InKernelPlanner, not reconstructed from text or logs.

## Lane

internal/agent and internal/gateway

## Suggested paths

- internal/agent/sample_options.go
- internal/agent/chat.go
- internal/agent/inkernel_decode.go
- internal/agent/inkernel_planner.go
- internal/agent/inkernel_measurement_test.go
- internal/gateway/wire.go
- internal/gateway/http.go
- internal/gateway/native_inference_receipt_test.go

## Acceptance

fak validate --mine internal/agent/sample_options.go internal/agent/chat.go internal/agent/inkernel_decode.go internal/agent/inkernel_planner.go internal/agent/inkernel_measurement_test.go internal/gateway/wire.go internal/gateway/http.go internal/gateway/native_inference_receipt_test.go

## Closure

One signed commit with Fixes metadata and a recognized fak leaf trailer.

## Parent context

#8972

## Why this is next

The frozen receipt requires 64 token ids, 64 finite selected-token scores, and true phase timing from the same guarded native request. Those facts exist inside InKernelPlanner but are not carried to the HTTP response, so a docs-only runner cannot produce valid evidence.

## Working spine

Explicit request opt-in -> SampleOpt -> real in-kernel decode observations -> typed Completion receipt -> ChatResponse fak extension -> deterministic HTTP readback.

## Outcome

A benchmark client can obtain a machine-readable native inference receipt from production fak serve without bypassing adjudication or reconstructing evidence from text.

## Dependencies

#9069 is required only for the long-context live campaign, not for this deterministic receipt implementation. Both unblock #8972.

## Work unit

leaf

## Scale

S1

## Expected steps

5

## Assumptions

The first supported scoring envelope is temperature zero with no logit bias, top-p truncation, or repetition penalties; the selected-token score is log_softmax of the actual unmodified model logits used for that choice.

## Coordination notes

Own only the named internal/agent and internal/gateway files. Do not touch #8972 docs, model/context sizing, or hardware state.

## Trigger

The #8972 native attempt reached production serve but no accepted native sample could be recorded because the response omitted token ids, per-step scores, and exact phase evidence.

## Batch policy

One issue and one commit for the buffered native receipt spine; streaming, external-provider logprobs, and full-vocabulary tensors remain separate leaves.

## Witness

- `go test ./internal/agent ./internal/gateway -run 'Test.*Native.*Receipt|Test.*InKernel.*Measurement' -count=1`

## Acceptance gate

`fak validate --mine internal/agent/sample_options.go internal/agent/chat.go internal/agent/inkernel_decode.go internal/agent/inkernel_planner.go internal/agent/inkernel_measurement_test.go internal/gateway/wire.go internal/gateway/http.go internal/gateway/native_inference_receipt_test.go`

## Likely files

- internal/agent/sample_options.go
- internal/agent/chat.go
- internal/agent/inkernel_decode.go
- internal/agent/inkernel_planner.go
- internal/agent/inkernel_measurement_test.go
- internal/gateway/wire.go
- internal/gateway/http.go
- internal/gateway/native_inference_receipt_test.go

## Model tier

- Required model tier: tier/T2-required
- Optimal model tier: tier/T2-optimal

## Closure binding

Close only from one signed commit that fixes #9070, carries a recognized fak leaf trailer, proves default-wire omission plus exact native receipt readback, and passes exact-path validation.

**Comment by anthony-chaudhary** [comment]
Claimed for implementation from current origin/main. Read-only trace confirms this is the smallest production-path prerequisite for #8972: InKernelPlanner already owns real decode logits and phase timings, but Completion and the gateway wire discard them. Exact scope is the issue-listed internal/agent and internal/gateway paths; the opt-in will reject unsupported sampling/streaming/provider paths, preserve default wire bytes, and carry no reconstructed identity. Dedupe checked: #4522/#4524 are quality-oracle overlap, not production serve receipt duplicates.

**Comment by anthony-chaudhary** [comment]
Implemented and committed the guarded native inference receipt spine on main as aab03e605d65. Focused witness is green: go test ./internal/agent ./internal/gateway -run 'Test.*Native.*Receipt|Test.*InKernel.*Measurement' -count=1. Isolated acceptance validation passed (fak validate --mine over the eight issue paths), gateway package test and vet passed, and dos commit-audit reports the commit witnessed. Publishing is temporarily blocked after the sanctioned sync adaptation: fak sync check --fetch found main diverged (+2/-2), and git merge --no-edit origin/main refused because peer-owned dirty docs/generated/disambiguation/canonical-terms.md and contrast-index.md would be overwritten. Smallest next checkable step: when that peer lands those two generated docs, merge origin/main in place and run fak sync push. Promotion evidence: explicit request opt-in reaches SampleOpt and deterministic HTTP readback; demotion/retirement evidence: default responses omit the extension and streaming/modified sampling refuse; invalidating assumption: selected-token log_softmax is valid only for the first supported greedy, unmodified sampling envelope.
